includes/functions.php

```diff
@@ -495,25 +495,21 @@ function isPingable($hostname,$device_id = FALSE)
    if(is_numeric($config['fping_options']['timeout']) || $config['fping_options']['timeout'] > 1) {
        $fping_params .= ' -t ' . $config['fping_options']['timeout'];
    }
-   $status = shell_exec($config['fping'] . "$fping_params -e $hostname 2>/dev/null");
+    if(is_numeric($config['fping_options']['count']) || $config['fping_options']['count'] > 0) {
+        $fping_params .= ' -c ' . $config['fping_options']['count'];
+    }
+   //$status = shell_exec($config['fping'] . "$fping_params -e $hostname 2>/dev/null");
    $response = array();
-   if (strstr($status, "alive"))
-   {
-     $response['result'] = TRUE;
+   $status = fping($hostname,$fping_params);
+   if ($status['loss'] < 0 || $status['loss'] == 100) {
```

Comment:
That's for detecting it's down, so if a device has 100% packet loss then it's down (if no value is returned this is due to a bad hostname so we class that as down).

my https://librenms.lathwood.co.uk branch is running this at present if you want a mix of up / down devices.